mlrun/db/httpdb.py

```diff
@@ -1144,6 +1176,162 @@ def list_artifacts(
         :param limit:           Maximum number of artifacts to return.
         """
 
+        artifacts, _ = self._list_artifacts(
+            name=name,
+            project=project,
+            tag=tag,
+            labels=labels,
+            since=since,
+            until=until,
+            iter=iter,
+            best_iteration=best_iteration,
+            kind=kind,
+            category=category,
+            tree=tree,
+            producer_uri=producer_uri,
+            format_=format_,
+            limit=limit,
+            return_all=True,
+        )
+        return artifacts
+
+    def paginated_list_artifacts(
+        self,
+        name: Optional[str] = None,
+        project: Optional[str] = None,
+        tag: Optional[str] = None,
+        labels: Optional[Union[str, dict[str, Optional[str]], list[str]]] = None,
+        since: Optional[datetime] = None,
+        until: Optional[datetime] = None,
+        iter: int = None,
+        best_iteration: bool = False,
+        kind: str = None,
+        category: Union[str, mlrun.common.schemas.ArtifactCategories] = None,
+        tree: str = None,
+        producer_uri: str = None,
+        format_: Optional[
+            mlrun.common.formatters.ArtifactFormat
+        ] = mlrun.common.formatters.ArtifactFormat.full,
+        limit: int = None,
+        page: Optional[int] = None,
+        page_size: Optional[int] = None,
+        page_token: Optional[str] = None,
+    ) -> tuple[ArtifactList, Optional[str]]:
+        """List artifacts with support for pagination and various filtering options.
```

Comment:
instead of doc-dup list_artifacts paramters, you can just set a reference to list_Artifacts doc_string. the examples are ok to be left.

also, instead of accepting named arguments, Id perhaps have the function being declared as
```
def paginated_list_artifacts(self, *args, page=None, ...., **kwargs):
  pass
```

please add example of iterating the pages automatically without the need to explicitly state the page